src/session/content/item_row.rs

```diff
@@ -94,16 +96,267 @@ impl ItemRow {
+                                            )
+                                        }
+                                    }
+                                ));
                             }
-                            _ => self.set_child_row::<MessageText>(message),
+                            MessageContent::MessageChatDeleteMember(data) => {
+                                self.get_or_create_event_row().set_label(&format!(
+                                    "<b>{}</b>",
+                                    match message.sender() {
+                                        MessageSender::User(user) if user.id() == data.user_id => {
+                                            if message.is_outgoing() {
+                                                gettext("You left the group")
+                                            } else {
+                                                gettext!("{} left the group", sender_name)
+                                            }
+                                        }
+                                        _ => {
+                                            let user_name = super::item_row::sender_name(
+                                                &message
+                                                    .chat()
+                                                    .session()
+                                                    .user_list()
+                                                    .get(data.user_id),
+                                            );
+                                            if message.is_outgoing() {
+                                                gettext!("You removed {} from the group", user_name)
+                                            } else {
+                                                gettext!(
+                                                    "{} removed {} from the group",
+                                                    sender_name,
+                                                    user_name
+                                                )
+                                            }
+                                        }
+                                    }
+                                ));
+                            }
+                            MessageContent::MessagePinMessage(data) => {
+                                let msg =
+                                    match message.chat().history().message_by_id(data.message_id) {
+                                        Some(data) => match data.content().0 {
+                                            MessageContent::MessageText(data) => {
+                                                let msg = data.text.text;
+                                                if msg.chars().count() > 17 {
+                                                    gettext!(
+                                                        " «{}...»",
```

Comment:
The 3 dots are only valid if @melix agrees. See comment below.
The other thing is: Was the whitespace intentional? You already have a whitespace in line 314 and  317. So I removed it to not have double whitespace.
```suggestion
                                                        "«{}…»",
```